File: RafJira/app/src/main/java/com/example/MainActivity.java
package com.example;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.splashscreen.SplashScreen;
import androidx.fragment.app.FragmentTransaction;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;

import com.example.constants.Constants;
import com.example.raf_jira.R;
import com.example.raf_jira.databinding.ActivityMainBinding;
import com.example.view.fragments.LoginFragment;
import com.example.view.fragments.MainFragment;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        SplashScreen splashScreen = SplashScreen.installSplashScreen(this);
        super.onCreate(savedInstanceState);

        splashScreen.setKeepOnScreenCondition(() -> {

            FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();

            SharedPreferences sharedPreferences = getSharedPreferences(Constants.SHARED_PREFERENCE_NAME, Context.MODE_PRIVATE);
            boolean isLogged = sharedPreferences.getBoolean(Constants.IS_LOGGED_KEY, false);
            if(isLogged)
                fragmentTransaction.replace(R.id.fragment_container_view_tag, new MainFragment(), Constants.MAIN_FRAGMENT_TAG);
            else
                fragmentTransaction.replace(R.id.fragment_container_view_tag, new LoginFragment(), Constants.LOGIN_FRAGMENT_TAG);

            fragmentTransaction.commit();
            return false;
        });
        getSupportActionBar().hide();
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);
    }
}

File: Dnevnjak/app/src/main/java/com/example/dnevnjak/MainActivity.kt
package com.example.dnevnjak

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
import com.example.dnevnjak.presentation.fragments.LoginFragment
import com.example.dnevnjak.utilities.Constants
import com.example.dnevnjak.utilities.Constants.Companion.LOGIN_FRAGMENT_TAG

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        installSplashScreen().apply {
            setKeepOnScreenCondition {

                val sharedPreferences = getSharedPreferences(Constants.SHARED_PREFERENCES_PATH, MODE_PRIVATE)
                val isLoggedBefore = sharedPreferences.getString(Constants.USERNAME_KEY, null)
                val transaction = supportFragmentManager.beginTransaction()

                if(isLoggedBefore.isNullOrEmpty())
                    transaction.replace(R.id.activity_main_fragment_container, LoginFragment(), LOGIN_FRAGMENT_TAG)
                else{

                }
                transaction.commit()

                false
            }
        }

        setContentView(R.layout.activity_main)

    }

}



